src/chatd.cpp

```diff
@@ -4329,7 +4360,100 @@ void Chat::handleTruncate(const Message& msg, Idx idx)
 
     // if truncate was received for a message not loaded in RAM, we may have more history in DB
     mHasMoreHistoryInDb = at(lownum()).id() != mOldestKnownMsgId;
+    truncateAttachmentHistory();
+}
+
+void Chat::handleRetentionTime()
+{
+    uint32_t retentionTime = getRetentionTime();
+    if (!retentionTime)
+    {
+        // If retentionTime is disabled
+        return;
+    }
+
+    // Get idx of the most recent msg affected by retention time, if any
+    chatd::Idx idx = CHATD_IDX_INVALID;
+    time_t ts = time(nullptr) - retentionTime;
+    CALL_DB(getIdxByRetentionTime, ts, idx);
+    if (idx == CHATD_IDX_INVALID)
+    {
+        // If there are no messages to remove
+        return;
+    }
+
+    Message *msg = findOrNull(idx);
+    if (msg)
+    {
+       // GUI must detach and free any resources associated with erased messages
+       CALL_LISTENER(onRetentionHistoryTruncated, *msg, idx, getMsgStatus(*msg, idx));
+    }
+
+    // Discard current key, if any
+    CALL_CRYPTO(resetSendKey);
+
+    // Clean affected messages in db and RAM
+    CHATID_LOG_DEBUG("Cleaning messages previous to %d seconds", retentionTime);
+    CALL_DB(retentionHistoryTruncate, idx);
+    truncateByRetentionTime(idx);
+
+    // Clean pending rich links
+    removePendingRichLinks(idx);
+
+    // update oldest index in db
+    mOldestIdxInDb = mDbInterface->getOldestIdx();
+
+    if (mOldestIdxInDb == CHATD_IDX_INVALID) // If there's no messages in db
+    {
+        mLastIdxReceivedFromServer = CHATD_IDX_INVALID;
+        mLastIdReceivedFromServer = karere::Id::null();
```

Comment:
As I can see in develop lastIdReceivedFromServer It's only set but not used for anymore, and the only one getter it's not used. If we don't use this var anymore could we remove it?

Reply:
It looks it might be handy for the management of the `RECEIVED` pointer, so the `mLastIdxReceivedFromServer`, since the `idx` should correspond to the `msgid`.
However, as you said, the feature is currently disabled and not working, so I think it's fine to remove it. In the future, if the `RECEIVED` is enabled (only for 1on1), we would see if the `msgid` is needed.